lightbar_manager/include/lightbar_manager/lightbar_manager_node.hpp

```diff
@@ -82,49 +98,70 @@ class LightBarManager
         * \brief Callback function for turning signal
         * \return 
         */
-        void turnSignalCallback(const automotive_platform_msgs::TurnSignalCommandPtr& msg_ptr);
+        void turnSignalCallback(automotive_platform_msgs::msg::TurnSignalCommand::UniquePtr msg_ptr);
+
+        ////
+        // Overrides
+        ////
+        carma_ros2_utils::CallbackReturn handle_on_configure(const rclcpp_lifecycle::State &);
+        carma_ros2_utils::CallbackReturn handle_on_activate(const rclcpp_lifecycle::State &);
 
     private:
+        
+        Config config_;
+
         /*!
-        * \brief Helper function that sets up ROS parameters for unit test
-        * \return none
+        * \brief Turn signal callback function helper
+        * \return 
         */
-        void setupUnitTest();
+        void processTurnSignal(const automotive_platform_msgs::msg::TurnSignalCommand& msg);
 
         // Node Data
-        std::string node_name_;
-        double spin_rate_;
+        std::string node_name_ = "lightbar_manager";
```

Comment:
I realized it was not needed in the worker class, but still needed on node level here. This is because the node uses this name to check its priority against potential other plugins